inference-explainers/inference_onboarding_guide.md

```diff
@@ -0,0 +1,228 @@
+*Note: Other versions of Tensorflow and PyTorch will be supported in the future.*
+
+#### Step 2: Deployment
+
+1. **Upload ML models:** Use a GCS bucket to host proprietary ML models. The Bidding server fetches the uploaded models during runtime and forwards them to the inference sidecar for handling requests. The bucket name is required by the Terraform configuration. Additionally, you must upload a model configuration file that specifies the models being fetched to the cloud bucket and include its path relative to the cloud bucket in the Terraform configuration. The model configuration file has the following format:
+
+    ```json
+    {
+      "model_metadata": [
+        {
+          "model_path": "model_1",
+          "checksum": "<SHA-256 checksum value of hexadecimal string>"
+        },
+        {
+          "model_path": "model_2",
+          "checksum": "<SHA-256 checksum value of hexadecimal string>"
+        }
+      ]
+    }
+    ```
+    Each model is specified by its path relative to the cloud bucket and may be accompanied by an optional SHA-256 checksum of the model's content. The checksum can be computed by using the following bash command:
+
+    ```
+    find <model_path> -type f -exec sha256sum {} \; | sort -k 2 | awk '{print $1}' | tr -d '\n' | sha256sum | awk '{print $1}'
+    ```
+
+    Store the metadata file in the same bucket as the models. You can update it after server deployment to fetch additional models. Note that you must grant READ permission for your proprietary endpoints to access your models and use them with your service.
+
+2. **Upload Code Modules:** The inference capabilities of your proprietary JavaScript bidding code modules are exposed with the `runInference` and `getModelPaths` inference callbacks. To use inference capabilities, update your existing code modules to invoke these functions. Refer to [this section][7] of the B&A Inference Overview Explainer for more details about these two functions, and to [this section][8] of the GCP Guide for more information about uploading code modules.
+
+3. **Configure Terraform variables:** Configure the inference capabilities using inference Terraform flags. The names of all inference-related Terraform flags have an `INFERENCE` prefix. Include them as part of the `runtime_flags` in the “buyer” module. Use `INFERENCE_SIDECAR_BINARY_PATH` to expose your ML runtime selection. To enable inference and select the ML runtime, set this flag to `/server/bin/inference_sidecar_<runtime_name>`, where `<runtime_name>` is either tensorflow_v2_14_0 for a Tensorflow runtime with the version 2.14.0, or pytorch_v2_1_1 for a PyTorch runtime with the version 2.1.1 . **Note that inference is disabled if this flag is not set.** If inference is disabled, all inference-related Terraform flags are ignored and invocations of inference callbacks in code modules will fail. <br/>
```

Comment:
should we break this up into 3.1 3.2 etc? This section is really big and needs some breaks ...

Reply:
Also move **Note that ...** to a different line. I _think_ starting with Note: will render the line in a note box

Reply:
Done for AWS and GCP